algebird-core/src/main/scala/com/twitter/algebird/CorrelationGroup.scala

```diff
@@ -0,0 +1,165 @@
+package com.twitter.algebird
+
+object Correlation {
+  def apply(x: (Double, Double), weight: Double): Correlation =
+    Correlation(c2 = 0, m2Left = 0, m2Right = 0, m1Left = x._1, m1Right = x._2, weight)
+
+  def apply(x: (Double, Double)): Correlation =
+    apply(x, 1.0)
+
+  implicit val group: Group[Correlation] = CorrelationGroup
+
+  /**
+   * When combining averages, if the counts sizes are too close we
+   * should use a different algorithm.  This constant defines how
+   * close the ratio of the smaller to the total count can be:
+   */
+  private val STABILITY_CONSTANT = 0.1
+
+  /**
+   * Given two streams of doubles (weightN, an) and (weightK, ak) of form (weighted count,
+   * mean), calculates the mean of the combined stream.
+   *
+   * Uses a more stable online algorithm which should be suitable for
+   * large numbers of records similar to:
+   * http://en.wikipedia.org/wiki/Algorithms_for_calculating_variance#Parallel_algorithm
+   *
+   * This differs from the implementation in MomentsGroup.scala only in that here, the counts are weighted, and are
+   * thus doubles instead of longs
+   */
+  def getCombinedMean(weightN: Double, an: Double, weightK: Double, ak: Double): Double =
```

Comment:
I don't love that this is copy-pasta. We could in principle refactor the method in `Moments` to take a `Numeric` or something like that, but I'm not sure if that would force you to bump a semantic version of Algebird, so I opted for a little bit of copying. Happy to take another approach if you prefer, @johnynek.